tests/w2w_test.py

```diff
@@ -1,50 +1,113 @@
 import os
 import pytest
-import rioxarray
+import rioxarray as rxr
 from affine import Affine
 import shutil
 from w2w.w2w import main
-from w2w.w2w import check_lcz_wrf_extent
+from w2w.w2w import _check_lcz_wrf_extent
+from w2w.w2w import _replace_lcz_number
+from w2w.w2w import check_lcz_integrity
 from w2w.w2w import wrf_remove_urban
 from w2w.w2w import create_wrf_gridinfo
 from w2w.w2w import calc_distance_coord
-import xarray
+import xarray as xr
+import numpy as np
 
 
 def test_argparse_shows_help():
     with pytest.raises(SystemExit):
         main(['--help'])
 
+def test_replace_lcz_number_ok(capsys):
+    info = {
+        'src_file': 'testing/Shanghai.tif',
+    }
+    LCZ_BAND = 0
+    lcz = rxr.open_rasterio(info['src_file'])[LCZ_BAND, :, :]
+
+    lcz_100 = np.array(
+        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+         101, 102, 103, 104, 105, 106, 107]
+    )
+    lcz_new = _replace_lcz_number(
+            lcz=lcz,
+            lcz_to_change=lcz_100,
+        )
+
+    # Test whether LCZs 100+ are converted to 11+
+    assert (np.unique(lcz_new.data).tolist() ==
+            [1, 2, 3, 4, 5, 6, 8, 10, 11, 12, 14, 15, 17]) \
+           is True
```

Comment:
the `is True` is not needed here since `np.unique(lcz_new.data).tolist() == [1, 2, 3, 4, 5, 6, 8, 10, 11, 12, 14, 15, 17]` already is a boolean. So `assert np.unique(lcz_new.data).tolist() == [1, 2, 3, 4, 5, 6, 8, 10, 11, 12, 14, 15, 17])` should be enough.

Reply:
What about the order of `np.unique(lcz_new.data).tolist()`  -- is it always sorted? otherwise this might be a flaky test, when the order randomly changes. Instead we could use a set for this

Reply:
np.unique always returns sorted values: https://numpy.org/doc/stable/reference/generated/numpy.unique.html.

I changed to assert ...